File: VB/CustomColorPIckEdit/RepositoryItemMyColorPickEdit.vb
Imports Microsoft.VisualBasic
Imports DevExpress.XtraEditors.Drawing
Imports DevExpress.XtraEditors.Registrator
Imports DevExpress.XtraEditors.Repository
Imports DevExpress.XtraEditors.ViewInfo
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Linq

Namespace Q535402
	Public Class RepositoryItemMyColorPickEdit
		Inherits RepositoryItemColorPickEdit
		Private _ShowClear As Boolean = True
		<DefaultValue(True)> _
		Public Property ShowClear() As Boolean
			Get
				Return _ShowClear
			End Get
			Set(ByVal value As Boolean)
				_ShowClear = value
			End Set
		End Property
		Public Const fEditorName As String = "MyColorPickEdit"
		Public Overrides ReadOnly Property EditorTypeName() As String
			Get
				Return fEditorName
			End Get
		End Property
		Public Sub New()
		End Sub
		Shared Sub New()
			RegisterEditor()
		End Sub
		Public Shared Sub RegisterEditor()
			EditorRegistrationInfo.Default.Editors.Add(New EditorClassInfo(fEditorName, GetType(MyColorPickEdit), GetType(RepositoryItemMyColorPickEdit), GetType(ColorEditViewInfo), New ColorEditPainter(), True, EditImageIndexes.ColorEdit))
		End Sub
		Public Overrides Sub Assign(ByVal item As RepositoryItem)
			BeginUpdate()
			Try
				MyBase.Assign(item)
				Dim source = TryCast(item, RepositoryItemMyColorPickEdit)
				If source Is Nothing Then
					Return
				End If
				ShowClear = source.ShowClear
			Finally
				EndUpdate()
			End Try
		End Sub
		Protected Overrides Sub RaisePopup(ByVal e As EventArgs)
			MyBase.RaisePopup(e)
		End Sub
		Public Overrides Sub RaiseColorPickDialogShowing(ByVal e As DevExpress.XtraEditors.ColorPickDialogShowingEventArgs)
			MyBase.RaiseColorPickDialogShowing(e)
		End Sub
	End Class
End Namespace

File: CS/CustomColorPIckEdit/RepositoryItemMyColorPickEdit.cs
// Developer Express Code Central Example:
// How to add a Clear button to implement a clear edit value functionality
// 
// This example demonstrates how to add a Clear button to the ColorPickEdit
// dropdown form to implement a clear edit value functionality.
// For this, add the
// ShowClear property to the RepositoryItemMyColorPickEdit class, create a
// SimpleButton, and add it to the Control collection of the dropdown form. The
// ShowClear property specifies the visibility of the Clear button in the dropdown
// form. If the value is true, the dropdown form contains a Clear button;
// otherwise, false. The default is true. The Clear button allows end-users to
// clear the currently selected color. Clicking the button assigns the null value
// to the edit value and closes the dropdown. You can use this approach both in
// inplace and standalone modes.
// 
// You can find sample updates and versions for different programming languages here:
// http://www.devexpress.com/example=E4943

using DevExpress.XtraEditors.Drawing;
using DevExpress.XtraEditors.Registrator;
using DevExpress.XtraEditors.Repository;
using DevExpress.XtraEditors.ViewInfo;
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Linq;

namespace Q535402 {
    public class RepositoryItemMyColorPickEdit : RepositoryItemColorPickEdit {
        private bool _ShowClear = true;
        [DefaultValue(true)]
        public bool ShowClear {
            get {
                return _ShowClear;
            }
            set {
                _ShowClear = value;
            }
        }
        public const string fEditorName = "MyColorPickEdit";
        public override string EditorTypeName {
            get {
                return fEditorName;
            }
        }
        public RepositoryItemMyColorPickEdit() {
        }
        static RepositoryItemMyColorPickEdit() {
            RegisterEditor();
        }
        public static void RegisterEditor() {
            EditorRegistrationInfo.Default.Editors.Add(new EditorClassInfo(fEditorName,
                typeof(MyColorPickEdit),
                typeof(RepositoryItemMyColorPickEdit),
                typeof(ColorEditViewInfo),
                new ColorEditPainter(), true, EditImageIndexes.ColorEdit));
        }
        public override void Assign(RepositoryItem item) {
            BeginUpdate();
            try {
                base.Assign(item);
                var source = item as RepositoryItemMyColorPickEdit;
                if (source == null) {
                    return;
                }
                ShowClear = source.ShowClear;
            } finally {
                EndUpdate();
            }
        }
        protected override void RaisePopup(EventArgs e) {
            base.RaisePopup(e);
        }
        public override void RaiseColorPickDialogShowing(DevExpress.XtraEditors.ColorPickDialogShowingEventArgs e) {
            base.RaiseColorPickDialogShowing(e);
        }
    }
}
